### ISSUE TITLE
dot-notation error when using ES6 Map class

### ISSUE THREAD

**Issue by axefrog** [babel/babel-eslint#68]
```js
var mymap = new Map();
mymap['foo'] = 'xyz';
```
yields:
```
error  ["foo"] is better written in dot notation  dot-notation
```

The `dot-notation` rule is valid for general cases, but not when using the `Map` class, and possibly should be the opposite, i.e. accessing keys should perhaps avoid dot notation in order to stay distinguished from explicit class properties such as `size`.

**Comment by eventualbuddha** [comment]
Isn't it `mymap.set('foo', 'xyz')`? I believe this rule still applies. If I'm wrong please point out the relevant part of the spec.

**Comment by axefrog** [comment]
You're absolutely right. My mistake.
